### PR TITLE
Fix GenotypeGVCFs with mixed ploidy sites

### PR DESCRIPTION
**Pull request by meganshand**
Currently if a site has too many alleles GenomicsDB doesn't output PLs for diploid samples at that site. At sites where all samples are diploid this site is successfully skipped. If the site has a mix of haploid and diploid calls (for example chrX with the latest versions of dragen on WGS data), then the site wasn't being skipped correctly and an error would be thrown by GenotypeGVCFs at the annotation step.

Now the site will be skipped if any of the called sites don't have PLs excluding no calls and hom ref genotypes. If the site is all no-calls or hom-ref then it retains the current behavior by only skipping sites where all the genotypes are missing PLs. I'm not sure if this happens frequently in the wild, but it is a common edge case in our tests.

### CODE REVIEW

**Comment by meganshand** [comment]
#carrot(HaplotypeCaller CARROT Regression Tests, VariantCallingCarrotOrchestrated.gatk_docker, )

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/GenotypingEngine.java
Comment (ldgauthier) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/GenotypingEngine.java:411]: Good catch!

File: src/test/java/org/broadinstitute/hellbender/tools/walkers/GenotypeGVCFsIntegrationTest.java
Comment (ldgauthier) [inline src/test/java/org/broadinstitute/hellbender/tools/walkers/GenotypeGVCFsIntegrationTest.java:1105]: I think I'd rather the test you added start from VCFs than commit a weird genomicsDB structure to the repo that might break if we update GenomicsDB (which admittedly doesn't happen that often).  Also the VCF is human-readable, whereas I have no idea what data you just added here. Take a look at testMaxAltsToCombineInGenomicsDB:
        final File tempGenomicsDB2 = GenomicsDBTestUtils.createTempGenomicsDB(inputs, interval);
        final String genomicsDBUri2 = GenomicsDBTestUtils.makeGenomicsDBUri(tempGenomicsDB2);

**Review by ldgauthier — COMMENTED** [review]
First off, thanks for jumping in here!  This is probably bus number one code, so it's great to get another set of eyes.

Can you reformat the test?  (There are some specifics inline.). I'm too lazy to check out the branch and dump the GenomicsDB, and I'd rather have human-readable test inputs anyway.

Also why are you running the Carrot tests for HaplotypeCaller?  I'm not convinced your changes will affect HaplotypeCaller.  I think it makes more sense to run WARP on jenkins.

**Comment by droazen** [comment]
The Carrot run failed due to PAPI error code 9, by the way, not for any reason specific to this branch:

```
                "executionStatus": "Failed",
                                "message": "Task BenchmarkComparison.EVALRuntimeTask:NA:4 failed. Job exit code 1. Check gs://dsde-methods-carrot-prod-cromwell/BenchmarkVCFsHeadToHeadOrchestrated/2aab3ed0-746b-451e-a7db-1c22fbb1bb29/call-CHMSampleHeadToHead/BenchmarkComparison/82289acc-83e7-49c8-acd0-9b2277166e10/call-EVALRuntimeTask/attempt-4/stderr for more information. PAPI error code 9. Please check the log file for more details: gs://dsde-methods-carrot-prod-cromwell/BenchmarkVCFsHeadToHeadOrchestrated/2aab3ed0-746b-451e-a7db-1c22fbb1bb29/call-CHMSampleHeadToHead/BenchmarkComparison/82289acc-83e7-49c8-acd0-9b2277166e10/call-EVALRuntimeTask/attempt-4/EVALRuntimeTask.log.",
                        "message": "Workflow failed"
    "status": "Failed",
                            "message": "Task BenchmarkComparison.EVALRuntimeTask:NA:4 failed. Job exit code 1. Check gs://dsde-methods-carrot-prod-cromwell/BenchmarkVCFsHeadToHeadOrchestrated/2aab3ed0-746b-451e-a7db-1c22fbb1bb29/call-CHMSampleHeadToHead/BenchmarkComparison/82289acc-83e7-49c8-acd0-9b2277166e10/call-EVALRuntimeTask/attempt-4/stderr for more information. PAPI error code 9. Please check the log file for more details: gs://dsde-methods-carrot-prod-cromwell/BenchmarkVCFsHeadToHeadOrchestrated/2aab3ed0-746b-451e-a7db-1c22fbb1bb29/call-CHMSampleHeadToHead/BenchmarkComparison/82289acc-83e7-49c8-acd0-9b2277166e10/call-EVALRuntimeTask/attempt-4/EVALRuntimeTask.log.",
                    "message": "Workflow failed"
            "message": "Workflow failed"
```

**Comment by meganshand** [comment]
@ldgauthier The test is much clearer now, thanks for pointing me to the example. This will end up being tested in WARP with the next GATK release and I'm not sure how easy it is to test two commits of GATK in WARP against each other. If it's possible to do that without updating the official truth data, then I could run that before we merge this. Otherwise we'll end up catching any issues when we update WARP after the next GATK release (which I'm motivated to do when the time comes).

**Comment by meganshand** [comment]
After speaking with Laura, I will test this commit in WARP against the previous commit to make sure that this change doesn't alter the JointGenotyping outputs. I also added another edge case test, but I want to make sure I run this version on a full size ~60 sample cohort of dragen generated WGS samples over chrX to make sure the downstream code works ok with other haploid/diploid mixes of no-calls/calls.

File: src/test/java/org/broadinstitute/hellbender/tools/walkers/GenotypeGVCFsIntegrationTest.java
Comment (ldgauthier) [inline src/test/java/org/broadinstitute/hellbender/tools/walkers/GenotypeGVCFsIntegrationTest.java:1130]: I'm feeling a little paranoid after so many GenotypeGVCFs hiccups.  If you've got the patience to wait for tests again, can you check that the remaining VC has genotypes called as expected with PLs?  We do expect PLs right?

**Review by ldgauthier — APPROVED** [review]
Tests are much easier to read now, thanks.  I'd like some genotype checks on the VC that does get emitted, but on the other hand other tests should probably cover that as well.

**Comment by meganshand** [comment]
I updated the test to check the output genotype here and I'm now in the process of running the WARP tests and on a dragen callset with 60 haploid/diploid mix samples.

**Comment by meganshand** [comment]
WARP tests for Exome single sample, WGS single sample, and JointGenotyping have all succeeded so I'm going to merge this one.

**Pull request merged**
